tsconfig.strict.json

```diff
@@ -5,101 +5,499 @@
     "noImplicitAny": true,
     "lib": ["es2020"]
   },
-  "include": [
-    "lib/config/app-strings.ts",
-    "lib/config/global.ts",
-    "lib/config/migrations/*",
-    "lib/config/presets/common.ts",
-    "lib/constants/**/*.ts",
-    "lib/data-files.generated.ts",
-    "lib/datasource/**/common.ts",
-    "lib/datasource/**/types.ts",
-    "lib/datasource/adoptium-java",
-    "lib/datasource/artifactory",
-    "lib/datasource/gitlab-tags/util.ts",
-    "lib/datasource/helm/common.ts",
-    "lib/datasource/metadata.ts",
-    "lib/datasource/sbt-plugin/util.ts",
-    "lib/globals.d.ts",
-    "lib/logger/**/*.ts",
-    "lib/manager/**/common.ts",
-    "lib/manager/**/types.ts",
-    "lib/manager/ansible-galaxy/util.ts",
-    "lib/manager/argocd/util.ts",
-    "lib/manager/gitlabci/utils.ts",
-    "lib/manager/gradle/deep/build-gradle.ts",
-    "lib/manager/gradle/shallow/tokenizer.ts",
-    "lib/manager/gradle/shallow/utils.ts",
-    "lib/manager/helm-values/util.ts",
-    "lib/manager/homebrew/util.ts",
-    "lib/manager/npm/extract/type.ts",
-    "lib/manager/npm/post-update/rules.ts",
-    "lib/manager/pre-commit/parsing.ts",
-    "lib/manager/terragrunt/providers.ts",
-    "lib/manager/terragrunt/util.ts",
-    "lib/platform/**/types.ts",
-    "lib/platform/github/graphql.ts",
-    "lib/platform/utils/pr-body.ts",
-    "lib/platform/utils/read-only-issue-body.ts",
-    "lib/proxy.ts",
-    "lib/types/**/*.ts",
-    "lib/util/**/*.ts",
-    "lib/versioning/gradle/**/*.ts",
-    "lib/versioning/cargo/**/*.ts",
-    "lib/versioning/hex/**/*.ts",
-    "lib/versioning/loose/**/*.ts",
-    "lib/versioning/maven/**/*.ts",
-    "lib/versioning/node/**/*.ts",
-    "lib/versioning/npm/**/*.ts",
-    "lib/versioning/pep440/**/*.ts",
-    "lib/versioning/rez/**/*.ts",
-    "lib/versioning/ruby/**/*.ts",
-    "lib/versioning/semver-coerced/**/*.ts",
-    "lib/versioning/semver/**/*.ts",
-    "lib/versioning/swift/**/*.ts",
-    "lib/versioning/ubuntu/**/*.ts",
-    "lib/workers/pr/changelog/hbs-template.ts",
-    "lib/workers/pr/changelog/types.ts",
-    "lib/workers/repository/init/types.ts",
-    "lib/workers/repository/model/commit-message.ts",
-    "test/exec-util.ts",
-    "test/graphql-snapshot.ts",
-    "test/http-mock.ts",
-    "test/json-schema.ts",
-    "test/newline-snapshot-serializer.ts",
-    "test/static-files.spec.ts",
-    "tools/utils/index.ts"
-  ],
+  "include": ["lib/", "test/"],
   "exclude": [
-    "lib/constants/platform.spec.ts",
-    "lib/datasource/adoptium-java/**/*.spec.ts",
-    "lib/datasource/artifactory/**/*.spec.ts",
+    "lib/**/*.spec.ts",
```

Comment:
Let's ignore tests until source files are fixed.

Lots of tests are using `index.ts` files from the top-level. It causes circular behaviour which is a little bit overwhelming when you're trying to quickly fix a single folder.